File: src/org/codepulsar/pulsar/Pulsar.java
package org.codepulsar.pulsar;

import java.io.IOException;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.nio.file.NoSuchFileException;
import java.nio.file.Paths;

public class Pulsar {
    public static void main(String[] args) {
        if (args.length == 0) {
            repl(); // TODO Implement REPL
        } else if (args.length == 1) {
            CommandsKt.parseCommands(args[0].trim());
        } else if (args.length == 2) {
            CommandsKt.parseCommands(args[0].trim(), args[1].trim());
        } else {
            CommandsKt.error("Invalid Parameters Given; Only 1 or 2 Parameters Allowed");
        }
    }

    private static void repl() {
    }

    public static void interpretFile(String file) throws IOException {
        Interpreter interpreter = new Interpreter(openFile(file));
        interpreter.interpret();
    }

    public static void compileFile(String file) throws IOException {
        Compiler compiler = new Compiler(openFile(file));
        compiler.init();
    }

    private static String openFile(String file) throws IOException {
        String stringFile;
        try {
            byte[] fileBytes = Files.readAllBytes(Paths.get(file));
            stringFile = new String(fileBytes, Charset.defaultCharset());
            return stringFile;
        } catch (NoSuchFileException fileException) {
            CommandsKt.error("The File Path That Was Provided Was Not Found");
            return null;
        }
    }
}

File: src/codepulsar/pulsar/Pulsar.cpp
#include "Pulsar.h"

Pulsar::SetUp conditions = {"", false, "0.1.0"};


int main(int argc, char* argv[]) {
    if (argc == 1) {
        Pulsar::repl();
    } else if (argc == 2) {
        Pulsar::parseCommands(argv[1]);
    } else if (argc == 3) {
        Pulsar::parseCommands(argv[1], argv[2]);
    } else {
        Pulsar::error("Invalid Commands Given: Only 1 Or 2 Parameters Are Allowed But " + std::to_string(argc) + " Parameters Were Given");
    }
}

void Pulsar::repl() {
}

void Pulsar::interpretFile(std::string file) {
    Pulsar::Interpreter interpreter = Pulsar::Interpreter(openFile(file));
    interpreter.interpret();
}

void Pulsar::compileFile(std::string file) {
    Pulsar::Compiler compiler = Pulsar::Compiler(openFile(file));
    compiler.init();
}

std::string Pulsar::openFile(std::string fileIn) {
    std::ifstream file;
    file.open(fileIn);

    std::string stringOut;
    std::string temp;

    if (file.is_open()) {
        while (getline(file, temp)) stringOut += temp + "\n";
        return stringOut;
    } else {
        error("Could Not Read File " + fileIn);
        return "";
    }
}
